framework/src/vectorpostprocessors/PointSamplerBase.C

```diff
@@ -159,10 +168,49 @@ PointSamplerBase::transferPointsVector(std::vector<Point> && points)
 const Elem *
 PointSamplerBase::getLocalElemContainingPoint(const Point & p)
 {
-  const Elem * elem = (*_pl)(p);
+  if (_discontinuous_variables)
+  {
+    // Get all possible elements the point may be in
+    std::set<const Elem *> candidate_elements;
+    (*_pl)(p, candidate_elements);
+
+    if (candidate_elements.begin() != candidate_elements.end())
+    {
+      // Look at all the element IDs
+      std::set<dof_id_type> candidate_ids;
+      std::transform(candidate_elements.begin(),
+                     candidate_elements.end(),
+                     std::inserter(candidate_ids, candidate_ids.end()),
+                     []( const Elem * elem ){ return elem->id(); });
```

Comment:
As much as I like stl methods, this still looks better IMO...

```suggestion
      std::set<dof_id_type> candidate_ids;
      for (const auto candidate : candidate_elements)
        candidate_ids.insert(candidate->id());
```

Reply:
I ll batch it in with the extra tests